### PR TITLE
Migrate to Gradle build system (opinionated)

### PR DESCRIPTION
**Pull request by janheinrichmerker**
This PR migrates the build from Maven to Gradle.
Besides better readable build config, this makes the build run on any machine, using the bundled, yet lightweight, Gradle wrapper.
You can read more about the benefits of Gradle here: https://gradle.org/maven-vs-gradle/

### CODE REVIEW

**Comment by janheinrichmerker** [comment]
After resolving some missing dependencies (repositories are [not transitive in Gradle](https://docs.gradle.org/current/userguide/declaring_repositories.html#strict_limitation_to_declared_repositories)), it now builds, as you can see here: https://jitpack.io/#heinrichreimer/trec-near-duplicates/d9aae6c2f9

**Comment by mam10eks** [comment]
I am open to change the build system to gradle, but I will need some time to review this merge request since the final, fat jar must be able to run in our spark-cluster. (and also some scripts need to be changed, e.g. the Makefile and minor changes in the scripts in src/main/bash)

**Comment by janheinrichmerker** [comment]
Ah, I see. Didn't know about any other build prerequisites.

**Comment by janheinrichmerker** [comment]
Now all dependencies from third-party/local repos are included in the release JAR.

File: Makefile [hunk 1]
```diff
@@ -145,10 +145,3 @@ clean-stuff:
 	rm -Rf /tmp/qrels* &&\
 	rm -Rf /tmp/tmp-unzipped* &&\
 	rm -Rf /tmp/.run*
-
-install-third-party: checkout-submodules
-	mvn -f third-party/jtreceval/ clean install
-
-checkout-submodules:
-	git submodule update --init --recursive
-
```
Comment (janheinrichmerker) [inline Makefile:154]: Those are not needed anymore, as there's no submodule to checkout and build.

File: Makefile [hunk 2]
```diff
@@ -129,7 +129,7 @@ evaluate-clueweb12: install
 	./src/main/bash/run-evaluation-report.sh -c CLUEWEB12 --threshold 0.84|tee results/clueweb12-evaluation.jsonl
 
 install: install-third-party
-	mvn clean install
+	./gradlew clean publishToMavenLocal
```
Comment (janheinrichmerker) [inline Makefile:132]: The `publishToMavenLocal` command builds the Gradle project and publishes the JAR artifact to the local Maven repo, effectively what `mvn install` did.

File: Makefile (hunk 1 above)
Comment (janheinrichmerker) [reply Makefile:154]: See #1 😃

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    implementation("org.slf4j:slf4j-log4j12:1.7.26")
+    implementation("com.jayway.jsonpath:json-path:2.4.0")
+    implementation("io.anserini:anserini:0.5.1")
+    implementation("com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.6.7")
+    implementation("org.elasticsearch.client:elasticsearch-rest-client:6.7.1")
+    implementation("client.netspeak:netspeak-client:1.3.5")
+    implementation("net.jodah:failsafe:2.0.1")
+    implementation("org.apache.solr:solr-core:8.1.1")
+    implementation("com.github.mam10eks:jtreceval:-SNAPSHOT")
+    implementation("org.apache.commons:commons-lang3:3.9")
```
Comment (janheinrichmerker) [inline build.gradle.kts:40]: See #4

**Comment by janheinrichmerker** [comment]
Closing due to inactivity

**Pull request closed without merging**